src/timeMachine/components/TagSelector.tsx

```diff
@@ -148,39 +126,60 @@ class TagSelector extends PureComponent<Props> {
       )
     }
 
+    let tag = selectedKey
+
+    if (tag === '' && index === 0) {
+      tag = '_measurement'
+    }
+
     const placeholderText =
       aggregateFunctionType === 'group'
         ? 'Search group column values'
-        : `Search ${selectedKey} tag values`
+        : `Search ${tag} tag values`
     return (
       <>
+        {!this.isCompliant && (
+          <BuilderCard.DropdownHeader
+            options={['filter', 'group']}
+            selectedOption={this.renderAggregateFunctionType(
+              aggregateFunctionType
+            )}
+            onDelete={index !== 0 && this.handleRemoveTagSelector}
+            onSelect={this.handleAggregateFunctionSelect}
+            isInCheckOverlay={this.props.isInCheckOverlay}
+          />
+        )}
         {this.isCompliant && <BuilderCard.Header title="Measurement" />}
         <BuilderCard.Menu testID={`tag-selector--container ${index}`}>
-          {aggregateFunctionType !== 'group' && (
-            <FlexBox
-              direction={FlexDirection.Row}
-              alignItems={AlignItems.Center}
-              margin={ComponentSize.Small}
-            >
-              <ErrorBoundary>
-                {!this.isCompliant && (
-                  <SearchableDropdown
-                    searchTerm={keysSearchTerm}
-                    emptyText="No Tags Found"
-                    searchPlaceholder="Search keys..."
-                    buttonStatus={toComponentStatus(keysStatus)}
-                    selectedOption={selectedKey}
-                    onSelect={this.handleSelectTag}
-                    onChangeSearchTerm={this.handleKeysSearch}
-                    testID="tag-selector--dropdown"
-                    buttonTestID="tag-selector--dropdown-button"
-                    menuTestID="tag-selector--dropdown-menu"
-                    options={keys}
-                  />
-                )}
-              </ErrorBoundary>
-              {this.selectedCounter}
-            </FlexBox>
+          {!this.isCompliant && (
```

Comment:
made this align with notebooks so that we have consistent UI across the app. Hopefully this can be better implemented with the new query builder experience by reusing component architecture